api/provider/twitter.go

```diff
@@ -40,6 +40,7 @@ type twitterUser struct {
 	Name      string `json:"name"`
 	AvatarURL string `json:"profile_image_url_https"`
 	Email     string `json:"email"`
+	ID        string `json:"id_str"`
```

Comment:
thoughts on standardizing these other ones to `Sub` to match Apple and Azure?

Reply:
Hmm, i'm not too sure if we should do that because the "Sub" name maps to the JWT "sub" claim. Both Apple & Azure support the openid connect protocol which return an `id_token` (jwt) that contains this claim. The rest of the providers use the OAuth2.0 Authorization Code flow and don't use this terminology.

Reply:
ah I was thinking that we were abstracting/standardizing AvatarUrl but it looks like we only do it for some and not all:

e.g. 
`AvatarURL string 'json:"picture"'`

vs `	AvatarURL string 'json:"profile_image_url_https"`

vs `Avatar`

maybe we should remove these other standardizations/abstractions? Doing it for some and not all fields might become a bit messy

Reply:
Hmm yeah the main issue is that the OAuth2.0 protocol doesn't have a standardised set of fields to provide information about a user. The OpenID Connect protocol was created to solve this problem by introducing a set of standardised claims. But each provider can still choose to add custom claims on top of the standard ones.

> maybe we should remove these other standardizations/abstractions?

This will be a breaking change because some devs rely on the metadata information we return [here](https://github.com/supabase/gotrue/blob/46df5fd65c6692635c672ce7c6811c069e6aa231/api/provider/twitter.go#L97). I can imagine that for devs using `sign-in with twitter` in their apps, they might get a user's profile pic based on the `raw_user_meta_data->>"avatar_url"` field. 

> Doing it for some and not all fields might become a bit messy

I think the most important fields we need to standardise are `provider_id`, `email` and `phone` (these should always be in the `provider_meta_data` column so we can create indices on the keys - `provider_meta_data->>'provider_id'`). The rest of the fields are provider-specific and can just be returned as they are without casting them into a struct.

Reply:
got it 👍